Detectors/ZDC/calib/src/BaselineCalib.cxx

```diff
@@ -0,0 +1,140 @@
+
+  if (mConfig == nullptr) {
+    LOG(fatal) << "o2::zdc::BaselineCalib: missing configuration object";
+    return -1;
+  }
+
+  if (mVerbosity > DbgZero) {
+    mModuleConfig->print();
+    mConfig->print();
+  }
+
+  if (opt.debug_output > 0) {
+    setSaveDebugHistos();
+  }
+
+  clear();
+  mInitDone = true;
+  return 0;
+}
+
+//______________________________________________________________________________
+void BaselineCalib::clear()
+{
+  mData.clear();
+}
+
+//______________________________________________________________________________
+int BaselineCalib::process(const o2::zdc::BaselineCalibSummaryData* data)
+{
+  if (!mInitDone) {
+    init();
+  }
+  if (mVerbosity >= DbgFull) {
+    data->print();
+  }
+  mData += data;
+  if (mVerbosity >= DbgFull) {
+    mData.print();
+  }
+  return 0;
+}
+
+//______________________________________________________________________________
+// Create calibration object
+int BaselineCalib::endOfRun()
+{
+  if (mVerbosity > DbgZero) {
+    LOGF(info, "Finalizing BaselineCalibData object");
+  }
+
+  // Compute average baseline
+  float factor = mModuleConfig->baselineFactor;
+  for (int ic = 0; ic < NChannels; ic++) {
+    double sum = 0;
+    double nsum = 0;
+    double bmin = mConfig->cutLow[ic];
+    double bmax = mConfig->cutHigh[ic];
+    for (int ib = 0; ib < BaselineRange; ib++) {
+      double bval = (BaselineMin + ib) * factor;
+      if (bval >= bmin && bval <= bmax) {
+        nsum += mData.mHisto[ic].mData[ib];
+        sum += bval * mData.mHisto[ic].mData[ib];
+      }
+    }
+    if (nsum > 0 && mConfig->min_e[ic]) {
+      float ave = sum / nsum;
+      LOGF(info, "Baseline %s %g events and cuts (%g:%g): %f", ChannelNames[ic].data(), nsum, bmin, bmax, ave);
+      mParamUpd.setCalib(ic, ave, true);
+    } else {
+      if (mParam == nullptr) {
+        LOGF(error, "Baseline %s %g events and cuts (%g:%g): CANNOT UPDATE AND MISSING OLD VALUE", ChannelNames[ic].data(), nsum, bmin, bmax);
+        mParamUpd.setCalib(ic, -std::numeric_limits<float>::infinity(), false);
+      } else {
+        float val = mParam->getCalib(ic);
+        LOGF(info, "Baseline %s %g events and cuts (%g:%g): %f NOT UPDATED", ChannelNames[ic].data(), nsum, bmin, bmax, val);
+        mParamUpd.setCalib(ic, val, false);
+      }
+    }
+  }
+
+  // Creating calibration object and info
+  auto clName = o2::utils::MemFileHelper::getClassName(mParamUpd);
+  mInfo.setObjectType(clName);
+  auto flName = o2::ccdb::CcdbApi::generateFileName(clName);
+  mInfo.setFileName(flName);
+  mInfo.setPath(CCDBPathBaselineCalib);
+  std::map<std::string, std::string> md;
+  md["config"] = mConfig->desc;
+  mInfo.setMetaData(md);
+  uint64_t starting = mData.mCTimeBeg;
+  if (starting >= 10000) {
+    starting = starting - 10000; // start 10 seconds before
+  }
+  uint64_t stopping = mData.mCTimeEnd + 10000; // stop 10 seconds after
+  mInfo.setStartValidityTimestamp(starting);
+  mInfo.setEndValidityTimestamp(stopping);
```

Comment:
you can set `mInfo.setAdjustableEOV()`, then every uploaded object will check for overlaps with previous and remove them if any.